src/coreclr/jit/fgbasic.cpp

```diff
@@ -849,7 +849,12 @@ void Compiler::fgFindJumpTargets(const BYTE* codeAddr, IL_OFFSET codeSize, Fixed
     const bool  isForceInline          = (info.compFlags & CORINFO_FLG_FORCEINLINE) != 0;
     const bool  makeInlineObservations = (compInlineResult != nullptr);
     const bool  isInlining             = compIsForInlining();
+    const bool  isTier1                = opts.jitFlags->IsSet(JitFlags::JIT_FLAG_TIER1);
+    const bool  resolveTokens          = makeInlineObservations && isTier1;
```

Comment:
This should be enabled for crossgen too.

Reply:
This covers the crossgen scenario, correct?
```suggestion
    const bool  isReadyToRun           = opts.IsReadyToRun();
    const bool  resolveTokens          = makeInlineObservations && (isTier1 || isReadyToRun);
```

Reply:
We usually use `opts.jitFlags->IsSet(JitFlags::JIT_FLAG_PREJIT)`

Reply:
Will switch to that. Could I get a summary of the difference between the two for future reference?

Reply:
```suggestion
    const bool  isPreJit               = opts.jitFlags->IsSet(JitFlags::JIT_FLAG_PREJIT);
    const bool  resolveTokens          = makeInlineObservations && (isTier1 || isPreJit);
```

Reply:
R2R is a sub-genre of prejitting, so we use `IsReadyToRun()` when we need to do something specifically for R2R, and use the flag check when we're doing something because we are prejitting.

In our own use the two coincide. Samsung still is using fragile NGEN (non-R2R prejitting) last I knew.

Reply:
Yep, as @AndyAyersMS said. PREJIT means AOT compilation in general. R2R means specific ABI for AOT compiled code, e.g. code sequence to access generic dictionaries, etc.